[EC-646] Org Admin Vault Refresh November Release Prep

shane-melton opened the pull request:
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Prepare the current OAVR work to be merged into `master` to prepare for the November release. Keeps the new `Billing`, `Settings`, `Reporting` tabs and brings back a modified version of the `Manage` tab (with only the Members, Collections, and Groups sub-pages). 

## Code changes

- **organization-layout.component.html:** Remove the Groups and Members tabs and replace with Manage
- **manage.component.html:** Remove side-nav links that are part of the OAVR being released. Keep the Members (renamed from People), Collections, and Groups links.
- **organization-routing.module.ts:** Add back the Manage tab routes
- **organization-billing.module.ts:** Add components that were missing from the billing module and was causing some missing functionality.
- **organization-settings.module.ts**: Remove components that were moved to org billing module.
- **import.component.html:** Fix import button. CSP changes broke the inline styles and JS that were being used originally
- **organization.service.abstraction.ts**: Re-introduce the Manage tab permission helper.
- **groups.component.html and people.component.html:** Remove the wrapping container class to prevent content overflow now that the tabs have a side nav again. (It looks like a giant number of line changes, but I only removed the wrapping div)

## Screenshots

<img width="1102" alt="image" src="https://user-images.githubusercontent.com/8764515/198414525-26ff654a-ce03-4787-a560-a9ed8620e672.png">

<img width="1035" alt="image" src="https://user-images.githubusercontent.com/8764515/198414689-359906df-2651-4ed3-a3c9-dd68111d7d68.png">


## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

coroiu commented on libs/common/src/abstractions/organization/organization.service.abstraction.ts:
```diff
@@ -24,6 +24,20 @@ export function canAccessBillingTab(org: Organization): boolean {
+  return canAccessMembersTab(org) || canAccessGroupsTab(org) || canManageCollections(org);
+}
+
 export function canAccessOrgAdmin(org: Organization): boolean {
```
I know you haven't changed anything in here but @jlf0dev found a bug in this method add added a fix to his own PR. But since we are not merging his PR into this release it might be nice to add just the fix?

The actually is just adding `org.isManager` at the bottom of the boolean expression.

coroiu replied:
Oh no wait!! Don't do this! I just found this

![image](https://user-images.githubusercontent.com/2285588/198519662-8476c33f-badd-4ab8-904b-67bed2c5c4e2.png)

coroiu commented on libs/components/src/tabs/shared/tab-list-item.directive.ts:
```diff
@@ -28,8 +28,23 @@ export class TabListItemDirective implements FocusableOption {
     return this.baseClassList
-      .concat(this.active ? this.activeClassList : ["!tw-text-main"])
-      .concat(this.disabled ? this.disabledClassList : []);
+      .concat(this.active ? this.activeClassList : [])
+      .concat(this.disabled ? this.disabledClassList : [])
+      .concat(this.textColorClassList);
+  }
+
+  /**
+   * Classes used for styling tab item text color.
+   * Separate text color class list required to override bootstrap classes in Web.
+   */
+  get textColorClassList(): string[] {
+    if (this.disabled) {
+      return ["!tw-text-muted", "hover:!tw-text-muted"];
+    }
+    if (this.active) {
+      return ["!tw-text-primary-500", "hover:!tw-text-primary-700"];
+    }
+    return ["!tw-text-main", "hover:!tw-text-main"];
   }
```
question: Is this just a refactor?

shane-melton replied:
Kind of, Chromatic was originally complaining about the disabled tab text color changing in this PR when its not. What actually happened is the text color was broken during a merge with `master` (Chromatic didn't see for a new "baseline" because there was no PR). I discovered this in the Groups PR, and fixed it there because Chromatic was complaining then as well. For the same reason, I needed to pull it in here to get Chromatic to stop saying there was a change. 

For context, this change is just making use of `!important` in for all the text classes, otherwise our bootstrap styles in Web will mess with the color because they're anchor tags underneath. We actually have a ticket tracking some of these temporary bootstrap workarounds in CL-30 (this change now included)

shane-melton replied:
@jlf0dev You also asked about this ☝️

coroiu reviewed (approved):
Guess it turned out to be a good thing that we didn't remove the unused components!
I'm approving the PR as-is, but the bug fix could be nice to get merged!

jlf0dev commented on libs/common/src/abstractions/organization/organization.service.abstraction.ts:
```diff
@@ -24,6 +24,20 @@ export function canAccessBillingTab(org: Organization): boolean {
+    org.canEditAnyCollection ||
+    org.canDeleteAnyCollection ||
+    org.canEditAssignedCollections ||
+    org.canDeleteAssignedCollections
```
You could use `canViewAssignedCollections` here if you wanted to simplify it. Just a suggestion.

jlf0dev reviewed (approved):
I'm also curious about Andreas' refactor question but other than that, LGTM!

jlf0dev reviewed (changes requested):
Sorry - Andreas' question about adding the `org.isManager` brings up a good point. We shouldn't add that check to the "Manage" check, but there was a bug introduced when we refactored permissions that still hides the "Organization" link in the header if the user is a manager. 

We need to add a a method to `organization.service.abstraction.ts`:
``` javascript
export function canAccessVaultTab(org: Organization): boolean {
  return org.isManager;
}
```

and then `canAccessOrgAdmin` becomes:

``` javascript
export function canAccessOrgAdmin(org: Organization): boolean {
  return (
    canAccessMembersTab(org) ||
    canAccessGroupsTab(org) ||
    canAccessReportingTab(org) ||
    canAccessBillingTab(org) ||
    canAccessSettingsTab(org) ||
    canAccessVaultTab(org)
  );
}
```

shane-melton commented:
@jlf0dev 

I've corrected the permission issues mentioned. Managers can now access the "Organization" link in the header now. Thanks for bringing this up!

@coroiu You still brought a good point, the manage tab should still show for managers in this version of OAVR as they need to view the collections page (since collection management is not yet under the Vaults tab). I also found that the tab was still trying to redirect to /manage/members for Managers, so that's now fixed.

jlf0dev reviewed (approved):
Good catch with the managers accessing the Manage Tab. Everything looks good to me!
